test/functional/combine_logs.py

```diff
@@ -78,10 +77,10 @@ def read_logs(tmp_dir):
     for each of the input log files."""
 
     # Find out what the folder is called that holds the debug.log file
-    chain = glob.glob("{}/node0/*/debug.log".format(tmp_dir))
-    if chain:
-        chain = chain[0]  # pick the first one if more than one chain was found (should never happen)
-        chain = re.search(r'node0/(.+?)/debug\.log$', chain).group(1)  # extract the chain name
+    # pick the first one if more than one chain was found (should never happen)
+    debugpath = next(pathlib.Path(tmp_dir).glob('node0/**/debug.log'),None)
+    if debugpath:
+        chain = re.search(r'node0/(.+?)/debug\.log$', debugpath.as_posix()).group(1)  # extract the chain name
```

Comment:
nit,

```py
glob = pathlib.Path(tmp_dir).glob('node0/**/debug.log')
path = next(glob, None)
if path:
    assert next(glob, None) == None      # should never happen
    chain = ...
    ...
```

Reply:
Done, that makes it a bit more readable. Also added assert.